homeassistant/components/transmission/__init__.py

```diff
@@ -331,44 +331,42 @@ def init_torrent_list(self):
 
     def check_completed_torrent(self):
         """Get completed torrent functionality."""
-        actual_torrents = self.torrents
-        actual_completed_torrents = [
-            var.name for var in actual_torrents if var.status == "seeding"
+        current_completed_torrents = [
+            torrent for torrent in self.torrents if torrent.status == "seeding"
         ]
-
-        tmp_completed_torrents = list(
-            set(actual_completed_torrents).difference(self.completed_torrents)
+        freshly_completed_torrents = set(current_completed_torrents).difference(
+            self.completed_torrents
```

Comment:
The old code would do set difference in terms of the `name` field.  (e.g. `self.completed_torrents`  assigned in `init_torrent_list` is based `x.name`).  This (simpler!) code is comparing `current_completed_torrents` which does not seem to look at `name` like it used to.

Is that OK?

Reply:
Thanks for noticing this.  In fact, a lot of this was confusing so I updated this PR to include type hinting.  I need to come back to this because some of the naming is still confusing.  But for this PR, even though it's starting to look like a lot, most of it is just prefixing _ on private attributes and adding type hints.

Reply:
So I think what you're saying is that it is intentional and is OK.  It is really just testing the objects themselves, which is probably better than testing on a field in the object.

Reply:
@allenporter It was broken but is fixed now.  I apologize for being unclear -  I was frustrated that some lists were objects and some lists were strings which is why I added the type-hinting.